src/fromager/sources.py

```diff
@@ -213,7 +218,7 @@ def download_url(
         return outfile
     # Open the URL first in case that fails, so we don't end up with an empty file.
     logger.debug(f"reading from {url}")
-    with requests.get(url, stream=True) as r:
+    with ctx.requests.get(url, stream=True) as r:
```

Comment:
How does retrying work with streaming? If it starts over, the caller needs to understand that so it doesn't just write duplicate copies of the data to the same file, right?

Reply:
Yeah, need to look at this

Reply:
Found this: https://stackoverflow.com/questions/25860105/python-requests-package-lost-connection-while-streaming but not sure whether it will help our cause

Reply:
https://gist.github.com/tobiasraabe/58adee67de619ce621464c1a6511d7d9 looks potentially useful. The `Range` HTTP header tells the server where to start sending data, based on how much has already been downloaded. https://medium.com/@lope.ai/downloading-files-over-http-with-python-requests-e12e6b795e43 shows another example of tracking that.

Reply:
Agreed
I have tried to use Retry header in the latest version of this PR